cache/cache.go

```diff
@@ -72,13 +74,32 @@ func New(watcher clientv3.Watcher, prefix string, opts ...Option) (*Cache, error
 		prefix:      prefix,
 		cfg:         cfg,
 		watcher:     watcher,
+		store:       newStore(),
 		ready:       make(chan struct{}),
 		stop:        cancel,
 		internalCtx: internalCtx,
 	}
 
 	cache.demux = NewDemux(internalCtx, &cache.waitGroup, cfg.HistoryWindowSize, cfg.ResyncInterval)
 
+	storeW := newWatcher(cfg.HistoryWindowSize, nil) // nil predicate == match all
+	cache.demux.Register(storeW, 0)
```

Comment:
You cannot register watch for storage on `0` revision because it's basically random. You need to register it on revision returned by get plus one.